sdks/python/apache_beam/examples/inference/sklearn_japanese_housing_regression.py

```diff
@@ -0,0 +1,161 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+"""A pipeline that uses RunInference API on a regression about housing prices.
+
+This example uses the japanese housing data from kaggle.
+
+Since the data has missing fields, this example illustrates how to split
+data and assign it to appropriate models. The predictions are then recombined.
+
+In order to set this example up, you will need two things.
+1. Build models (or use ours) and reference those via the model directory.
+2. Download the data from kaggle and host it.
+"""
+
+import argparse
+from typing import Iterable
+
+import apache_beam as beam
+from apache_beam.io.filesystems import FileSystems
+from apache_beam.ml.inference.base import RunInference
+from apache_beam.ml.inference.sklearn_inference import ModelFileType
+from apache_beam.ml.inference.sklearn_inference import SklearnModelHandlerPandas
+from apache_beam.options.pipeline_options import PipelineOptions
+from apache_beam.options.pipeline_options import SetupOptions
+import pandas
+
+MODELS = [{
+    'name': 'all_features',
+    'required_features': [
+        'Area',
+        'Year',
+        'MinTimeToNearestStation',
+        'MaxTimeToNearestStation',
+        'TotalFloorArea',
+        'Frontage',
+        'Breadth',
+        'BuildingYear'
+    ]
+},
+          {
+              'name': 'floor_area',
+              'required_features': ['Area', 'Year', 'TotalFloorArea']
+          },
+          {
+              'name': 'stations',
+              'required_features': [
+                  'Area',
+                  'Year',
+                  'MinTimeToNearestStation',
+                  'MaxTimeToNearestStation'
+              ]
+          }, {
+              'name': 'no_features', 'required_features': ['Area', 'Year']
+          }]
+
+
+def sort_by_features(dataframe, max_size):
+  """ Partitions the dataframe by what data it has available."""
+  for i, model in enumerate(MODELS):
+    required_features = dataframe[model['required_features']]
+    if not required_features.isnull().any().any():
+      return i
+  return -1
+
+
+def load_dataframe(file_name: str) -> Iterable[pandas.DataFrame]:
```

Comment:
Can we use a method instead of `DoFn` object for a `ParDo` that will achieve the same performance? just curious

From the examples I saw, `beam.ParDo()` is taking a `DoFn` object

Reply:
Here, it says it needs `DoFn` object. So may be we will change to DoFn class

https://beam.apache.org/documentation/programming-guide/#:~:text=When%20you%20apply%20a%20ParDo%20transform%2C%20you%E2%80%99ll%20need%20to%20provide%20user%20code%20in%20the%20form%20of%20a%20DoFn%20object.%20DoFn%20is%20a%20Beam%20SDK%20class%20that%20defines%20a%20distributed%20processing%20function.